processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -1123,6 +1129,24 @@ func (c *WatchClient) getReplicaSet(uid string) (*ReplicaSet, bool) {
 	return nil, false
 }
 
+// runInformer starts the given informer. This method optionally takes a list of other informers that should complete
+// before the informer is started. This is necessary e.g. for the pod informer which requires the replica set informer
+// to be finished to correctly establish the connection to the replicaset/deployment it belongs to.
+func (c *WatchClient) runInformer(informer cache.SharedInformer, dependencies ...cache.SharedInformer) {
```

Comment:
How about calling this one `runInformerWithDependencies` and having it receiving a `cache.InformerSynced` array directly?